### DISCUSSION TITLE
Why is there no Selected.Relationships?

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#654] [🙋♀️TE3 Q&A] [answered]
Hello, 

I have a script that goes through the relationships of the model by iterating on Model.Relationships and that works fine. 
Hoever, now I wanted to do a script on the selected relationships only, and to my surprise intellisense does not allow it.

There is something called Selected.SingleColumnRelationships which looks decent, but in the properties you can only get FromTable and ToTable, but no FromColumn or ToColumn 

Is there a reason for that? 
It would be nice to be able to go Selected.Relationships as we can for Model. 
Any recommendation to get the desired effect?

**Comment by otykier** [comment]
`Selected.SingleColumnRelationships` is the way to go.

![image](https://user-images.githubusercontent.com/8976200/189518726-b826b3d5-0106-4e1d-afb4-3dcf094d77e9.png)

**Comment by bernatagulloesbrina** [comment]
That's odd i don't see the r.FromColumn in visual studio 2022

[image: image.png]

am I missing a project reference here?
This is the environment I'm using


using TabularEditor.TOMWrapper;
using TabularEditor.Scripting;



namespace TEscripts
{

    public class Class1
    {
        Model Model;
        TabularEditor.UI.UITreeSelection Selected;

        void script2()
        {
            CalculationGroupTable cg = Model.AddCalculationGroup("Model");

            foreach(Relationship r in Selected.SingleColumnRelationships)
            {
                if r.From
            }
        }

    }
}

**Comment by bernatagulloesbrina** [comment]
Nevermind!

I had to change the variable type to SingleColumnRelationship too

now it's working

El dom, 11 sept 2022 a las 11:22, Bernat Agulló (<
***@***.***>) escribió:

> That's odd i don't see the r.FromColumn in visual studio 2022
>
> [image: image.png]
>
> am I missing a project reference here?
> This is the environment I'm using
>
>
> using TabularEditor.TOMWrapper;
> using TabularEditor.Scripting;
>
>
>
> namespace TEscripts
> {
>
>     public class Class1
>     {
>         Model Model;
>         TabularEditor.UI.UITreeSelection Selected;
>
>         void script2()
>         {
>             CalculationGroupTable cg = Model.AddCalculationGroup("Model");
>
>             foreach(Relationship r in Selected.SingleColumnRelationships)
>             {
>                 if r.From
>             }
>         }
>
>     }
> }
>
>
>
> El dom, 11 sept 2022 10:33, Daniel Otykier ***@***.***>
> escribió:
>
>> Selected.SingleColumnRelationships is the way to go.
>>
>> [image: image]
>> <https://user-images.githubusercontent.com/8976200/189518726-b826b3d5-0106-4e1d-afb4-3dcf094d77e9.png>
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/TabularEditor/TabularEditor3/discussions/654#discussioncomment-3618027>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AJUUXHINXJYAC7S6VD4QN5DV5WKOBANCNFSM6AAAAAAQJAFPAM>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> <TabularEditor/TabularEditor3/repo-discussions/654/comments/3618027@
>> github.com>
>>
>

-- 

Bernat Agulló

Partner & Data Enthusiast

***@***.***

+34 637 93 42 76

**Comment by ricky7uio** [comment]
Hi Daniel, 

what if I want to get an annotation from the selected relationship? 

I'm getting this error:

```
Object	Type	Message	Line	Column
	Error	'List<SingleColumnRelationship>' does not contain a definition for 'GetAnnotation' and no accessible extension method 'GetAnnotation' accepting a first argument of type 'List<SingleColumnRelationship>' could be found (are you missing a using directive or an assembly reference?)	7	11
```

Thanks!

**Reply by otykier** [reply, marked as answer]
You have to iterate through the collection:

```csharp
foreach(var relationship in Selected.SingleColumnRelationships)
{
    var value = relationship.GetAnnotation("test");
    ...
}
```

...or if you know that the selection only contains 1 relationship:

```csharp
Selected.SingleColumnRelationships.Single().GetAnnotation(...);
```

**Reply by ricky7uio** [reply]
Thank you very much Daniel!
